client/my-sites/media-library/index.jsx

```diff
@@ -60,6 +60,7 @@ module.exports = React.createClass( {
 
 	componentWillUnmount: function() {
 		delete this.onSearch;
+		MediaActions.setLibrarySelectedItems( this.props.site.ID, [] );
```

Comment:
This breaks any gallery creation, since the selection is cleared when we go away from list view.
<img width="1057" alt="screen shot 2017-03-24 at 12 24 38 pm" src="https://cloud.githubusercontent.com/assets/1270189/24305998/ec81fdfe-108c-11e7-82d6-fee0e261b8d4.png">